iroh/src/endpoint.rs

```diff
@@ -965,69 +984,16 @@ impl Endpoint {
-    /// ```no_run
-    /// use iroh::{Endpoint, Watcher};
-    ///
-    /// # let rt = tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap();
-    /// # rt.block_on(async move {
-    /// let mep = Endpoint::builder().bind().await.unwrap();
-    /// let _relay_url = mep.home_relay().initialized().await;
-    /// # });
-    /// ```
-    pub fn home_relay(&self) -> impl n0_watcher::Watcher<Value = Vec<RelayUrl>> + use<> {
-        self.msock.home_relay()
-    }
-
-    /// Returns a [`Watcher`] for the direct addresses of this [`Endpoint`].
+    /// Waits for the node to be considered online.
     ///
-    /// The direct addresses of the [`Endpoint`] are those that could be used by other
-    /// iroh nodes to establish direct connectivity, depending on the network
-    /// situation. The yielded lists of direct addresses contain both the locally-bound
-    /// addresses and the [`Endpoint`]'s publicly reachable addresses discovered through
-    /// mechanisms such as [QAD] and port mapping.  Hence usually only a subset of these
-    /// will be applicable to a certain remote iroh node.
+    /// This currently means at least one relay server was connected,
+    /// and at least one local IP address is available.
     ///
-    /// The [`Endpoint`] continuously monitors the direct addresses for changes as its own
-    /// location in the network might change.  Whenever changes are detected this stream
-    /// will yield a new list of direct addresses.
-    ///
-    /// When issuing the first call to this method the first direct address discovery might
-    /// still be underway, in this case the [`Watcher`] might not be initialized with [`Some`]
-    /// value yet.  Once the first set of local direct addresses are discovered the [`Watcher`]
-    /// will always return [`Some`] set of direct addresses immediately, which are the most
-    /// recently discovered direct addresses.
-    ///
-    /// # Examples
-    ///
-    /// To get the first set of direct addresses use [`Watcher::initialized`]:
-    /// ```no_run
-    /// use iroh::{Endpoint, Watcher as _};
-    ///
-    /// # let rt = tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap();
-    /// # rt.block_on(async move {
-    /// let ep = Endpoint::builder().bind().await.unwrap();
-    /// let _addrs = ep.direct_addresses().initialized().await;
-    /// # });
-    /// ```
-    ///
-    /// [QAD]: https://www.ietf.org/archive/id/draft-ietf-quic-address-discovery-00.html
-    pub fn direct_addresses(&self) -> n0_watcher::Direct<Option<BTreeSet<DirectAddr>>> {
-        self.msock.direct_addresses()
+    /// This has no timeout, so if that is needed, you need to wrap it in a timeout.
+    pub async fn online(&self) {
+        let mut watch1 = self.msock.home_relay();
+        let mut watch2 = self.msock.direct_addresses();
+        tokio::join!(watch1.initialized(), watch2.initialized());
```

Comment:
I think this needs a cfg for wasm_browser to only wait on the relay (direct_addresses() never completes in the browser IIRC)